test/unittests.cpp

```diff
@@ -607,40 +821,54 @@ void TestG1MultiExpExpected()
     };
     vector<g1> bases = {one, one};
     g1 expected, result;
-    expected = one.mulScalar<1>({5});
-    result = g1::multiExp(bases, scalars).value();
+    expected = one.scale<1>({5});
+    result = g1::weightedSum(bases, scalars);
     if(!expected.equal(result))
     {
-        throw invalid_argument("TestG1MultiExpExpected: bad multi-exponentiation");
+        throw invalid_argument("TestG1WeightedSumExpected: bad multi-exponentiation");
     }
 }
 
-void TestG1MultiExpBatch()
+void TestG1WeightedSumBatch()
 {
-    g1 one = g1::one();
-    int64_t n = 1000;
-    vector<array<uint64_t, 4>> scalars;
-    vector<g1> bases;
-    // scalars: [s0,s1 ... s(n-1)]
-    // bases: [P0,P1,..P(n-1)] = [s(n-1)*G, s(n-2)*G ... s0*G]
-    for(int64_t i = 0, j = n-1; i < n; i++, j--)
-    {
-        scalars.insert(scalars.begin(), array<uint64_t, 4>{static_cast<uint64_t>(rand()%100000), 0, 0, 0});
-        bases.push_back(g1::zero());
-        bases[i] = one.mulScalar(scalars[0]);
-    }
-    // expected: s(n-1)*P0 + s(n-2)*P1 + s0*P(n-1)
-    g1 expected, tmp;
-    for(int64_t i = 0; i < n; i++)
-    {
-        tmp = bases[i].mulScalar(scalars[i]);
-        expected = expected.add(tmp);
-    }
-    g1 result = g1::multiExp(bases, scalars).value();
-    if(!expected.equal(result))
-    {
-        throw invalid_argument("bad multi-exponentiation");
-    }
+    const auto doTest = [](int64_t n) {
+        g1 one = g1::one();
+        vector<array<uint64_t, 4>> scalars;
+        vector<g1> bases;
+        // scalars: [s0,s1 ... s(n-1)]
+        // bases: [P0,P1,..P(n-1)] = [s(n-1)*G, s(n-2)*G ... s0*G]
+        for(int64_t i = 0; i < n; i++)
+        {
+            scalars.insert(scalars.begin(), array<uint64_t, 4>{static_cast<uint64_t>(rand()%100000), 0, 0, 0});
```

Comment:
why `insert` instead of `emplace_back()`?

Reply:
just copied from existing test code... let me double check what it's doing...